src/output-json-ftp.c

```diff
@@ -60,41 +60,35 @@ typedef struct LogFTPLogThread_ {
     MemBuffer          *buffer;
 } LogFTPLogThread;
 
-static json_t *JsonFTPLogCommand(Flow *f, FTPTransaction *tx)
+static void JsonFTPLogCommand(Flow *f, FTPTransaction *tx, JsonBuilder *jb)
 {
-    json_t *cjs = json_object();
-    if (!cjs) {
-        return cjs;
-    }
-
     /* Preallocate array objects to simplify failure case */
-    json_t *js_resplist = NULL;
-    json_t *js_respcode_list = NULL;
+    JsonBuilder *js_resplist = NULL;
+    JsonBuilder *js_respcode_list = NULL;
     if (!TAILQ_EMPTY(&tx->response_list)) {
-        js_resplist = json_array();
-        js_respcode_list = json_array();
+        js_resplist = jb_new_array();
+        js_respcode_list = jb_new_array();
 
         if (unlikely(js_resplist == NULL || js_respcode_list == NULL)) {
-            if (js_resplist) {
-                json_decref(js_resplist);
-            } else {
-                json_decref(js_respcode_list);
-            }
-            return cjs;
+            goto fail;
         }
     }
-
-    json_object_set_new(cjs, "command", json_string(tx->command_descriptor->command_name));
+    jb_set_string(jb, "command", tx->command_descriptor->command_name);
     uint32_t min_length = tx->command_descriptor->command_length + 1; /* command + space */
     if (tx->request_length > min_length) {
-        json_object_set_new(cjs, "command_data",
-                            JsonAddStringN((const char *)tx->request + min_length,
-                                           tx->request_length - min_length));
+        char *s = BytesToString(tx->request + min_length,
```

Comment:
@jasonish iirc we talked about a direct `jb_set_bytes` or something to avoid an expensive operation like this?

Reply:
`set_string_from_bytes` ... I'll update

Reply:
`jb_set_string_from_bytes` should work, just looks like it doesn't have an extern "C" wrapper yet (haven't needed it from C yet would be the reason).

Reply:
See this PR: https://github.com/OISF/suricata/pull/5040

Should be able to:
```
jb_append_string_from_bytes(js_respcode_list, where, 3);
```